test/e2e/utils/generate_incident.go

```diff
@@ -0,0 +1,211 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	sdk "github.com/PagerDuty/go-pagerduty"
+)
+
+const (
+	ClusterHasGoneMissing                         = "ClusterHasGoneMissing"
+	ClusterProvisioningDelay                      = "ClusterProvisioningDelay"
+	ClusterMonitoringErrorBudgetBurnSRE           = "ClusterMonitoringErrorBudgetBurnSRE"
+	InsightsOperatorDown                          = "InsightsOperatorDown"
+	MachineHealthCheckUnterminatedShortCircuitSRE = "MachineHealthCheckUnterminatedShortCircuitSRE"
+	ApiErrorBudgetBurn                            = "ApiErrorBudgetBurn"
+	AlertManagerDown                              = "AlertManagerDown"
+
+	// PagerDuty Events API endpoint
+	PagerDutyEventsURL = "https://events.pagerduty.com/v2/enqueue"
+)
+
+// EventPayload represents the payload structure for PagerDuty Events API
+type EventPayload struct {
+	Summary   string                 `json:"summary"`
+	Source    string                 `json:"source"`
+	Severity  string                 `json:"severity"`
+	Timestamp string                 `json:"timestamp"`
+	Details   map[string]interface{} `json:"custom_details,omitempty"`
+}
+
+// Event represents the complete event structure for PagerDuty Events API
+type Event struct {
+	RoutingKey  string        `json:"routing_key"`
+	EventAction string        `json:"event_action"`
+	DedupKey    string        `json:"dedup_key,omitempty"`
+	Payload     *EventPayload `json:"payload"`
+}
+
+// EventResponse represents the response from PagerDuty Events API
+type EventResponse struct {
```

Comment:
Already exists in the pagerduty sdk: https://pkg.go.dev/github.com/PagerDuty/go-pagerduty#EventResponse